src/lib/features/project-status/project-lifecycle-summary-read-model.test.ts

```diff
@@ -0,0 +1,112 @@
+import { addDays } from 'date-fns';
+import dbInit, { type ITestDb } from '../../../test/e2e/helpers/database-init';
+import getLogger from '../../../test/fixtures/no-logger';
+import { ProjectLifecycleSummaryReadModel } from './project-lifecycle-summary-read-model';
+import type { StageName } from '../../types';
+import { randomId } from '../../util';
+
+let db: ITestDb;
+
+beforeAll(async () => {
+    db = await dbInit('project_lifecycle_summary_read_model_serial', getLogger);
+});
+
+afterAll(async () => {
+    if (db) {
+        await db.destroy();
+    }
+});
+
+const updateFeatureStageDate = async (
+    flagName: string,
+    stage: string,
+    newDate: Date | null,
+) => {
+    if (newDate === null) {
+        return;
+    }
+
+    await db
+        .rawDatabase('feature_lifecycles')
+        .where({ feature: flagName, stage: stage })
+        .update({ created_at: newDate });
+};
+
+describe('Average time calculation', () => {
+    test('it calculates the average time for each stage', async () => {
+        const project1 = await db.stores.projectStore.create({
+            name: 'project1',
+            id: 'project1',
+        });
+        const now = new Date();
+
+        const flags = [
+            { name: randomId(), offsets: [2, 5, 6, 10] },
+            { name: randomId(), offsets: [1, null, 4, 7] },
+            { name: randomId(), offsets: [12, 25, 8, 9] },
+            { name: randomId(), offsets: [1, 2, 3, null] },
+        ];
+
+        for (const { name, offsets } of flags) {
+            const created = await db.stores.featureToggleStore.create(
+                project1.id,
+                {
+                    name,
+                    createdByUserId: 1,
+                },
+            );
+            await db.stores.featureLifecycleStore.insert([
+                {
+                    feature: name,
+                    stage: 'initial',
+                },
+            ]);
+
+            const stagesToAdd = ['pre-live', 'live', 'completed', 'archived'];
+
+            for (const index of [0, 1, 2, 3]) {
```

Comment:
Lets use foreach built in index so it is more tightly coupled to stage list.

Reply:
Used an array.entries in a68f05773f224d8ad4ef721054dfd84323841b43 and refined it in 0c69c275c6b2fd33ac81cba4981b22c017864b6b. 

`forEach` runs into problems with async, so this felt like the cleanest way to do it.